easypy/randutils.py

```diff
@@ -60,34 +60,36 @@ def load_dictionary(filename):
 VERBS = load_dictionary('verbs.txt')
 ADVERBS = load_dictionary('adverbs.txt')
 ADJECTIVES = load_dictionary('adjectives.txt')
-PARTS_OF_SPEECH = (NOUNS, VERBS, ADVERBS, ADJECTIVES)
+PARTS_OF_SPEECH = (ADVERBS, VERBS, ADJECTIVES, NOUNS)
 
 
 def random_meaningful_name(max_length=50, sep='-'):
     parts = []
-    remaining = max_length
+    next_max_part_len = max_length
 
     if max_length < 3:
         return random_string(max_length, charset=string.ascii_lowercase)
 
-    for part_group in PARTS_OF_SPEECH:
-        max_part_len = remaining - len(sep) if parts else max_length
+    # Reverse iterate so that partial sentences will sound logical
+    for part_group in PARTS_OF_SPEECH[::-1]:
```

Comment:
Because if the requested order is `adverb_verb_adjective_noun` partial sentences won't be logical if parts are iterated in order. We will get `adverb` or `adverb_verb_adjective`. Iterating backwards makes all possible partial sentences logical (apart from weird words chosen by random due to the extreme broadness of the dictionary).